servers/jm_utils/jm_utils/page_tokens.py

```diff
@@ -1,44 +1,137 @@
 import base64
+import datetime
 import json
 import numbers
 
+# We implement the pagination token via base64-encoded JSON s.t. tokens are
```

Comment:
nit: perhaps move into package comment instead (`"""` on the first line of the file)?

Reply:
Yup that's better.